### PR TITLE
fix: delete model version modal should dismiss on first try [WEB-744]

### PR DESCRIPTION
**Pull request by mapmeld**
## Description

As shown in the ticket, the **model version** delete modal cannot be canceled or closed without a second click. This change makes it more in line with our other modals.

## Test Plan

Cannot unregister another's model version
- **Sign in as determined** on a live version. Visit http://localhost:3000/det/models/333 or the ModelDetails page for another model which was not created by your user (determined)
- On the versions' right action menu, select "Deregister Version". The modal reads "Only the item creator or an admin can delete this item."
- In the modal, click the [x] button in top right to immediately dismiss
- Re-open the modal and click "Okay" to immediately dismiss
- Re-open the modal and press Esc [escape key] to immediately dismiss 

Can dismiss modal
- Create a test model and register an experiment checkpoint as a version inside this model.
- Visit the ModelDetails page
- On the versions' right action menu, select "Deregister Version". The modal reads "Are you sure you want to delete this version ... from this model?"
- In the modal, click the [x] button in top right to immediately dismiss
- Re-open the modal and click "Okay" to immediately dismiss
- Re-open the modal and press Esc [escape key] to immediately dismiss 

Can delete model version
- On the versions' right action menu, select "Deregister Version". The modal reads "Are you sure you want to delete this version ... from this model?"
- In the modal, click the red "Delete Version" button to confirm. A few seconds later, this version should disappear

## Checklist

- [x] Changes have been manually QA'd
- [x] User-facing API changes need the "User-facing API Change" label.
- [x] Release notes should be added as a separate file under `docs/release-notes/`.
  See [Release Note](https://github.com/determined-ai/determined/blob/master/docs/release-notes/README.md) for details.
- [x] Licenses should be included for new code which was copied and/or modified from any external code.

### CODE REVIEW

**Review by keita-determined — APPROVED** [review]
LGTM
we should hide `Deregister` option if a user doesnt have permission to deregister model version at some point (no need to be in this PR).

**Comment by mapmeld** [comment]
I agree on hiding Deregister. Just made a commit which does this.
- `CANNOT_DELETE_MODAL_PROPS` was used only by DeleteModel and DeleteModelVersion; it pre-dates our RBAC work where we remove actions from the UI
- Model delete can be accessed from ModelRegistry and ModelDetails/ModelHeader. In both we already checked `canDeleteModel` before displaying delete option
- Model Version delete can be accessed from ModelDetails and ModelVersionDetails/ModelVersionHeader. I changed both of these to check`canDeleteModelVersion`.

File: webui/react/src/shared/hooks/useModal/useModal.tsx [hunk 1]
```diff
@@ -49,16 +49,6 @@ interface ModalOptions {
  * If you skip the clone step, when attempting to re-open the
  * same modal to show the "cannot delete" message will fail.
  */
```
Comment (keita-determined) [inline webui/react/src/shared/hooks/useModal/useModal.tsx:51]: can this comment be removed with `CANNOT_DELETE_MODAL_PROPS`?

File: webui/react/src/hooks/useModal/Model/useModalModelVersionDelete.tsx [hunk 2]
```diff
@@ -54,24 +46,22 @@ const useModalModelVersionDelete = ({ onClose }: Props = {}): ModalHooks => {
...
-            onOk: () => handleOk(modelVersion),
-            title: 'Confirm Delete',
-          }
-        : clone(CANNOT_DELETE_MODAL_PROPS);
+        icon: null,
+        maskClosable: true,
+        okButtonProps: { type: 'primary' },
+        okText: 'Delete Version',
+        okType: 'danger',
+        onOk: () => handleOk(modelVersion),
```
Comment (keita-determined) [inline webui/react/src/hooks/useModal/Model/useModalModelVersionDelete.tsx:60]: nit: `handleOk` func can be inside `getModalProps` func, then we can remove dependencies and easier to follow the code imo

**Review by keita-determined — COMMENTED** [review]
mostly LGTM but there is one thing about model header
if a user doesn't have permission, we should hide `...` button too

https://user-images.githubusercontent.com/109113616/210436815-518d98a0-accb-48cb-8497-82f5b0b8e5c0.mov

File: webui/react/src/shared/hooks/useModal/useModal.tsx (hunk 1 above)
Comment (mapmeld) [reply webui/react/src/shared/hooks/useModal/useModal.tsx:51]: ✅ , also `DEFAULT_MODAL_PROPS` is used only in this file so I will move those props

File: webui/react/src/hooks/useModal/Model/useModalModelVersionDelete.tsx (hunk 2 above)
Comment (mapmeld) [reply webui/react/src/hooks/useModal/Model/useModalModelVersionDelete.tsx]: ✅

**Review by keita-determined — APPROVED** [review]
LGTM

prob we should make a ticket for this

https://user-images.githubusercontent.com/109113616/210440010-3e5c97b4-38be-4d91-af52-3991f9e15eb3.mov

**Comment by mapmeld** [comment]
I'll just hid that now in the latest commit

File: webui/react/src/pages/ModelVersionDetails/ModelVersionHeader.tsx [hunk 3]
```diff
@@ -221,11 +226,13 @@ my_model.load_state_dict(ckpt['models_state_dict'][0])`;
                 {action.text}
               </Button>
             ))}
-            <Dropdown menu={menu} trigger={['click']}>
-              <Button type="text">
-                <Icon name="overflow-horizontal" size="tiny" />
-              </Button>
-            </Dropdown>
+            {actions.length > 2 && (
```
Comment (keita-determined) [inline webui/react/src/pages/ModelVersionDetails/ModelVersionHeader.tsx:229]: this should be more then 1 instead of 2?

Comment (mapmeld) [reply webui/react/src/pages/ModelVersionDetails/ModelVersionHeader.tsx:229]: the actions should always include Download and Use in Notebook, is that wrong?

Comment (keita-determined) [reply webui/react/src/pages/ModelVersionDetails/ModelVersionHeader.tsx:229]: depends on the screen size

https://user-images.githubusercontent.com/109113616/210443761-a2db53b8-d603-4908-bfa8-b318d0958a80.mov
